snuba/processor.py

```diff
@@ -90,7 +90,7 @@ def _unicodify(s):
     if isinstance(s, dict) or isinstance(s, list):
         return json.dumps(s)
 
-    return six.text_type(s)
+    return six.text_type(s).encode('utf8', errors='backslashreplace').decode('utf8')
```

Comment:
Is this the fix for the utf8 issue ?
If yes, just to double check, it seems to me what goes into the replacement topic goes through the processor first, so there should be no fix to apply there, right ?

Reply:
> Is this the fix for the utf8 issue ?

Yep, this ensures what is assumed to be UTF-8 is actually UTF-8.

> If yes, just to double check, it seems to me what goes into the replacement topic goes through the processor first, so there should be no fix to apply there, right ?

This is a good point. I think this is the case (at least for new events) but I've always felt a little hazy on that stuff, so I'll double check. If the replacement data goes back through Python (I forget if we were able to optimize that out or not), there's the chance that existing data is not valid UTF-8 and would fail during merges (but not deletes) to be written back with the new state.

Reply:
> If the replacement data goes back through Python (I forget if we were able to optimize that out or not), there's the chance that existing data is not valid UTF-8 and would fail during merges (but not deletes) to be written back with the new state.

We were able to optimize this out with [`INSERT INTO … SELECT`](https://github.com/getsentry/snuba/blob/1073a7810452cae0ba9494d9a5f180d46ab24d73/snuba/replacer.py#L221-L225) so these shouldn't go back through Python. The replacer also still uses the native client so no worries there either.